odo/backends/bcolz.py

```diff
@@ -82,6 +101,22 @@ def resource_bcolz(uri, dshape=None, expected_dshape=None, **kwargs):
         dshape = datashape.dshape(dshape)
 
         dt = dshape_to_numpy(dshape)
+        if isrecord(dshape.measure):
+            dtype_pairs = [(name, dt[name].type) for name in dt.names]
+            is_object = [dtype == np.object_ for _, dtype in dtype_pairs]
+            if any(is_object):
+                raise TypeError("object dtypes are not supported by bcolz.\n"
```

Comment:
Well, object dtypes are actually supported, but I agree that they are actually not very usable unless the objects are large, in which case they are quite fine. Maybe you can change that message by "small object dtypes are poorly supported by bcolz"?